homeassistant/components/bayesian/config_flow.py

```diff
@@ -0,0 +1,208 @@
+    SchemaCommonFlowHandler,
+    SchemaConfigFlowHandler,
+    SchemaFlowFormStep,
+    SchemaFlowMenuStep,
+)
+
+from .const import (
+    CONF_P_GIVEN_F,
+    CONF_P_GIVEN_T,
+    CONF_PRIOR,
+    CONF_PROBABILITY_THRESHOLD,
+    CONF_TO_STATE,
+    DEFAULT_NAME,
+    DEFAULT_PROBABILITY_THRESHOLD,
+    DOMAIN,
+)
+
+
+class Observation_types(StrEnum):
+    """StrEnum for all the different observation types."""
+
+    STATE = "state"
+    NUMERIC_STATE = "numeric_state"
+    TEMPLATE = "template"
+
+    @staticmethod
+    def list() -> list[str]:
+        """Return a list of the values."""
+
+        return [c.value for c in Observation_types]
+
+
+async def _validate_mode(
+    handler: SchemaCommonFlowHandler, user_input: dict[str, Any]
+) -> dict[str, Any]:
+    """Validate the threshold mode, and set limits to None if not set."""
+    return {**user_input}
+
+
+OPTIONS_SCHEMA = vol.Schema(
+    {
+        vol.Required(
+            CONF_PROBABILITY_THRESHOLD, default=DEFAULT_PROBABILITY_THRESHOLD * 100
+        ): selector.NumberSelector(
+            selector.NumberSelectorConfig(
+                mode=selector.NumberSelectorMode.SLIDER,
+                step=1.0,
+                min=0,
+                max=100,
+                unit_of_measurement="%",
+            ),
+        ),
+        vol.Required(
+            CONF_PRIOR, default=DEFAULT_PROBABILITY_THRESHOLD * 100
+        ): selector.NumberSelector(
+            selector.NumberSelectorConfig(
+                mode=selector.NumberSelectorMode.SLIDER,
+                step=1.0,
+                min=0,
+                max=100,
+                unit_of_measurement="%",
+            ),
+        ),
+        vol.Optional(CONF_DEVICE_CLASS): selector.SelectSelector(
+            selector.SelectSelectorConfig(
+                options=[cls.value for cls in BinarySensorDeviceClass],
+                mode=selector.SelectSelectorMode.DROPDOWN,
+                translation_key="binary_sensor_device_class",
+                sort=True,
+            ),
+        ),
+    }
+)
+SUBSCHEMA_BOILERPLATE = vol.Schema(
+    {
+        vol.Required(CONF_P_GIVEN_T): selector.NumberSelector(
+            selector.NumberSelectorConfig(
+                mode=selector.NumberSelectorMode.SLIDER,
+                step=1.0,
+                min=0,
+                max=100,
+                unit_of_measurement="%",
+            ),
+        ),
+        vol.Required(CONF_P_GIVEN_F): selector.NumberSelector(
+            selector.NumberSelectorConfig(
+                mode=selector.NumberSelectorMode.SLIDER,
+                step=1.0,
+                min=0,
+                max=100,
+                unit_of_measurement="%",
+            ),
+        ),
+    }
+)
+
+NUMERIC_STATE_SUBSCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_ENTITY_ID): selector.EntitySelector(
+            selector.EntitySelectorConfig(
+                domain=[SENSOR_DOMAIN, INPUT_NUMBER_DOMAIN, NUMBER_DOMAIN]
```

Comment:
Are these the only domains that can support numeric values?

Reply:
Should be